### PR TITLE
Add support for MIPS32

### PR DESCRIPTION
**Pull request by rracariu**
Adds version fixes from upstream, fixes wrong mask size for mips32, fixes asm warning.

### CODE REVIEW

**Comment by redstar** [comment]
LGTM.

File: std/math.d [hunk 1]
```diff
@@ -5035,9 +5035,14 @@ private:
             }
             else version (MIPS_Any)
             {
-                cast(void) __asm!uint("cfc1 $0, $$31\n" ~
-                                      "andi $0, $0, 0xFFFFFF80\n" ~
-                                      "ctc1 $0, $$31", "=r");
+                version (D_LP64)    enum mask = "0xFFFFFF80";
+                else                enum mask = "0xFF80";
+
+                cast(void) __asm!uint(`.set noat
```
Comment (JohanEngelen) [inline std/math.d:5041]: The original code used `"..."~"..."~...` which results in better asm output text alignment. I'd rather keep that with minor legibility cost here.

Comment (rracariu) [reply std/math.d:5041]: There are other places in druntime using this style, I found it more pleasing to the eye than the old style.

I think you find yourself reading more of this D code than the ASM output of it, it makes it easier to understand for newbies also.

But I'm fine changing it back if this is the consensus.

**Comment by rracariu** [comment]
Ping. What's the verdict here? need to rollback to the original asm style? Thanks.

**Comment by JohanEngelen** [comment]
@rracariu It's OK. Just checking whether the change was made consciously, I don't have a strong opinion. More important to get your work in!

**Comment by JohanEngelen** [comment]
@rracariu To test Phobos changes, you have to submit an LDC PR to update the submodule. It's cumbersome, but it's all we have at the moment. (and you can only submit a submodule update for commits that are in the `ldc-developers/phobos` repo...)

**Pull request merged**